src/Dont/DontClone.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont;
+
+use Dont\Exception\NonCloneableObject;
+use Dont\Exception\TypeError;
+
+trait DontClone
```

Comment:
This is really nice, but the trait should be named `Clone` so the FQNS is `Dont\Clone` as per issue 3.